shell/platform/android/test/README.md

```diff
@@ -0,0 +1,95 @@
+# Unit testing Java code
+
+All Java code in the engine should now be able to be tested with Robolectric 3.8
+and JUnit 4. The test suite has been added after the bulk of the Java code was
+first written, so most of these classes do not have existing tests. Ideally code
+after this point should be tested, either with unit tests here or with
+integration tests in other repos.
+
+## Adding a new test
+
+1. Create a file under `test/` matching the path and name of the class under
+   test. For example,
+   `shell/platform/android/io/flutter/util/Preconditions.java` ->
+   `shell/platform/android/**test**/io/flutter/util/Preconditions**Test**.java`.
+2. Add your file to the `sources` of the `robolectric_tests` build target in
+   `/shell/platform/android/BUILD.gn`. This compiles the test class into the
+   test jar.
+3. Add your class to the `@SuiteClasses` annotation in `FlutterTestSuite.java`.
+   This makes sure the test is actually executed at run time.
+4. Write your test.
+5. Build and run with `testing/run_tests.py [--type=java] [filter=<test_class_name>]`.
+
+## Q&A
+
+### Why are we using Robolectric 3.8 when Robolectric 4+ is current?
+
+Robolectric 4+ uses the AndroidX libraries, and the engine sources use the
```

Comment:
@mklim @xster @blasten @amirh do we know what we're waiting for at this point to upgrade the engine dependencies? The plugin ecosystem is upgraded, right?

Reply:
As far as I'm aware we should be clear to migrate the engine too once flutter/flutter#23586 is fixed. @blasten has almost got the fix landed. The plugins are migrated but the final Jetifier fix still needs to be landed.